File: 1221_balance_string/balance.js
function balanceString(s) {
    let count = 0;
    let balance = 0;

    for(let l of s) {
        if(l === "R") count++;
        if(l === "L") count--;
        if(count === 0) balance++;
    }

    return balance;
}

console.log(balanceString("RLRRLLRLRL"))

File: 1221_balance_string/balance.py
def balanceString(s):
    count = 0
    balance = 0

    for l in s:
        if l == "R":
            count += 1
        if l == "L":
            count -= 1
        if count == 0:
            balance += 1

    return balance

print(balanceString("RLRRLLRLRL"))